### PR TITLE
GEOS-8464 Geofence: complex rules GUI

### PR DESCRIPTION
**Pull request by NielsCharlier**

### CODE REVIEW

File: src/community/geofence-server/src/main/java/org/geoserver/geofence/web/LayerAttributeModel.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package org.geoserver.geofence.web;
```
Comment (nmco) [inline src/community/geofence-server/src/main/java/org/geoserver/geofence/web/LayerAttributeModel.java]: Missing copyright header.

**Comment by nmco** [comment]
Just had a very quick look, the documentation update seems to be missing, will this be contributed in another PR ?

**Comment by etj** [comment]
In the tab "Layer details" the field "Layer type" should be automatically detected from the selected layer:
image

**Comment by etj** [comment]
In the PR implementation the layer attributes should be typed, while they could be automatically got from the selected featuretype and selected through a dropdown menu, but this can be considered a "nice to have":
image

**Comment by etj** [comment]
Bad error message:
image

**Comment by NielsCharlier** [comment]
okay, changes were made and docs added. I changed the approach to the layer attributes quite a bit, rather than a drop-down box all attributes are shown in the table with the option to specify access mode.

**Comment by NielsCharlier** [comment]
@etj can we merge this?

**Comment by NielsCharlier** [comment]
I have merged since all the reviewer's requests were fulfilled.

**Pull request merged**